### ISSUE TITLE
GIL detection not working

### ISSUE THREAD

**Issue by maartenbreddels** [benfred/py-spy#329]
Hi,

awesome tool! Just one issue that bugs me often is that GIL usage shows 0%.

Python 3.7 on linux, tested with py-spy 0.3.3 and 0.4.0.dev1

```python
from multiprocessing.pool import ThreadPool

def add(x):
    x + 1

pool = ThreadPool(100)
for i in range(1000):
    pool.map(add, range(1_000_000))
```
Running with `py-spy top -n -- python issues/pyspygil.py`  gives:
```

Collecting samples from 'python issues/pyspygil.py' (python v3.7.8)
Total Samples 2600
GIL: 0.00%, Active: 103.00%, Threads: 104

  %Own   %Total  OwnTime  TotalTime  Function (filename:line)                                            
  9.00%   9.00%   14.62s    14.62s   do_futex_wait.constprop.0 (libpthread-2.31.so)
 40.00%  80.00%   11.20s    19.73s   mapstar (multiprocessing/pool.py:44)
 37.00%  37.00%    7.18s     7.35s   add (pyspygil.py:4)
  4.00%   4.00%    1.33s     1.39s   worker (multiprocessing/pool.py:134)
```
The do_futex_wait gives it away (only visible with -n), but I am often confused by seeing GIL at 0%. 

Related #100 #104

**Comment by benfred** [comment]
Thanks for the bug report!

This is probably related to the version of python you are running - python 3.7 moved the gil location to a private api (pyruntime.gilstate.tstate_current) which is much harder to figure out from a separate program. In python 3.6 and older we could just load the _PyThreadState_Current symbol to get this - but getting the GIL in python 3.7+ has proven to be quite a bit harder. We're currently hardcoding the tstate_current location offset relative to the base interpreter  - but this can change with each major release, and is different for each OS and processor combo.  This solution is a bit of a hack, but I can't find a better way of doing this right now.

#338 adds support for detecting the GIL on python 3.7.8/3.7.9

**Comment by benfred** [comment]
@maartenbreddels This fix is in the latest release (v0.3.4) - can you try it out if you have a chance?

Btw, I really enjoyed your [post on tracing the python gil](https://www.maartenbreddels.com/perf/jupyter/python/tracing/gil/2021/01/14/Tracing-the-Python-GIL.html) - nice work!

**Comment by maartenbreddels** [comment]
Yes, 0.3.4 works 🎉 



> Btw, I really enjoyed your [post on tracing the python gil](https://www.maartenbreddels.com/perf/jupyter/python/tracing/gil/2021/01/14/Tracing-the-Python-GIL.html) - nice work!

Thanks!
